packages/studioCMS/src/integrations/studioCMSDashboard/routes/databaseSetup/main.astro

```diff
@@ -0,0 +1,190 @@
+              size="medium"
+              help-text="Would you like to setup a local Admin User or OAuth User?"
+              ><div class="textlabelinput"><sl-icon name="person-exclamation"></sl-icon> Setup Local Admin User?</div>
+            </sl-checkbox>
+
+            <sl-details id="local-admin" disabled="true">
+              <div class="textlabelinput" slot="summary"><sl-icon name="person-add"></sl-icon> Local Admin Setup</div>
+              <sl-input 
+              name="local-admin-name"
+              id="local-admin-name">
+                <div class="textlabelinput" slot="label">Username for Primary Admin <i>( Just set one, You can add more later )</i></div>
+              </sl-input><br />
+              <sl-input 
+              name="local-admin-display-name"
+              id="local-admin-display-name">
+                <div class="textlabelinput" slot="label">Displayname for Primary Admin</div>
+              </sl-input><br />
+              <sl-input 
+              type="password" 
+              name="local-admin-password"
+              id="local-admin-password"
+              password-toggle>
+                <div class="textlabelinput" slot="label">Password</div>
+              </sl-input>
+            </sl-details>
+
+            <sl-details id="oauthadmin" open>
+              <div class="textlabelinput" slot="summary"><sl-icon name="person-rolodex"></sl-icon> oAuth Admin Setup</div>
+              <sl-input 
+              name="oauth-admin-name"
+              id="oauth-admin-name"
+              placeholder="Username of Primary Admin (ex. GithubUsername)">
+                <div class="textlabelinput" slot="label">Default Site Admin <i>( Just set one, You can add more later )</i></div>
+              </sl-input>
+            </sl-details><br />
+
+
+            <div style="text-align:center">
+              <sl-button variant="success" type="submit" outline>Initialize Database!<div slot="suffix" class="i-mdi:login"></div></sl-button>
+            </div>
+
+          </form>
+
+          </div>
+        </div> 
+
+      <sl-alert id="form-error" variant="danger" duration="4000" closable>
+        <sl-icon slot="icon" name="exclamation-octagon"></sl-icon>
+        <strong>Something went wrong!</strong><br />
+        <div id = "form-error-message"></div>
+      </sl-alert>
+      <sl-alert id="form-success" variant="success" duration="4000" closable>
+        <sl-icon slot="icon" name="check2-circle"></sl-icon>
+        <strong>Success!</strong><br />
+        <div id = "form-success-message"></div>
+      </sl-alert>
+
+
+  </div>
+</Layout>
+
+<script>
+	const alertElement = document.getElementById("form-error")!;
+  const errorMessageElement = document.getElementById("form-error-message")!;
+	const successAlertElement = document.getElementById("form-success")!;
+  const successMessageElement = document.getElementById("form-success-message")!;
+  const checkbox = document.getElementById('local-setup')!;
```

Comment:
Does shoelace not export types for these components????

Reply:
is there a way to link types to a script imported element without transitioning to something like `@astojs/lit` to run the elements?

Reply:
Can't this be handled with type assertions?